### PR TITLE
Make datatype h5_model for machine leaning models

### PR DESCRIPTION
**Pull request by qiagu**
## What did you do? 
- [describe the proposed changes]
- Add a new datatype h5_model

## Why did you make this change?
(Cite Issue number OR provide rationalization of changes if no issue exists)
(If fixing a bug, please add any relevant error or traceback)

## How to test the changes? 
(select the most appropriate option; if the latter, provide steps for testing below)
- [x] I've included appropriate [automated tests](https://docs.galaxyproject.org/en/latest/dev/writing_tests.html).
- [ ] This is a refactoring of components with existing test coverage.
- [ ] Instructions for manual testing are as follows:
  1. [add testing steps and prerequisites here if you didn't write automated tests covering all your changes]

## License
- [x] I agree to license these contributions under [Galaxy's current license](https://github.com/galaxyproject/galaxy/blob/dev/LICENSE.txt).
- [x] I agree to allow the Galaxy committers to license these and all my past contributions to the core galaxy codebase under the [MIT license](https://opensource.org/licenses/MIT). If this condition is an issue, uncheck and just let us know why with an e-mail to galaxy-committers@lists.galaxyproject.org.

## For UI Components
- [x] I've included a screenshot of the changes

### CODE REVIEW

**Comment by qiagu** [comment]
Peek the model
image

File: lib/galaxy/config/sample/datatypes_conf.xml.sample [hunk 1]
```diff
@@ -170,6 +170,7 @@
     <datatype extension="hivtrace" type="galaxy.datatypes.text:Json" mimetype="application/json" subclass="true" display_in_upload="false"/>
     <datatype extension="cool" type="galaxy.datatypes.binary:Cool" mimetype="application/octet-stream" display_in_upload="true"/>
     <datatype extension="mcool" type="galaxy.datatypes.binary:MCool" mimetype="application/octet-stream" display_in_upload="true"/>
+    <datatype extension="h5_model" type="galaxy.datatypes.binary:H5_Model" mimetype="application/octet-stream" display_in_upload="true"/>
```
Comment (bgruening) [inline lib/galaxy/config/sample/datatypes_conf.xml.sample]: @qiagu my concern is that this name is too generic. Can we not name it gxmlm or gxml_model or something like that.

File: lib/galaxy/datatypes/binary.py [hunk 2]
```diff
@@ -1452,6 +1459,77 @@ def display_peek(self, dataset):
             return "MCool (HDF5) file (%s)." % (nice_size(dataset.get_size()))
 
 
+class H5_Model(H5):
```
Comment (jmchilton) [inline lib/galaxy/datatypes/binary.py]: I think ``H5Model`` would be a more consistent class name.

File: lib/galaxy/config/sample/datatypes_conf.xml.sample (hunk 1 above)
Comment (qiagu) [reply lib/galaxy/config/sample/datatypes_conf.xml.sample]: How about `h5mlm`? `h5` is more specific.

File: lib/galaxy/datatypes/binary.py (hunk 2 above)
Comment (qiagu) [reply lib/galaxy/datatypes/binary.py]: Thanks. Will update it once the name is finalized.

File: lib/galaxy/config/sample/datatypes_conf.xml.sample (hunk 1 above)
Comment (bgruening) [reply lib/galaxy/config/sample/datatypes_conf.xml.sample]: Also ok :)

**Comment by qiagu** [comment]
The hyperparameter block could be as large as a few KBs. Do you think it's a good idea to store it in metadata directly instead of a file (FileParameter)?

File: lib/galaxy/tool_util/xsd/galaxy.xsd [hunk 3]
```diff
@@ -3519,6 +3519,11 @@ used to generate dynamic options.
         <xs:documentation xml:lang="en">Documentation for startswith</xs:documentation>
       </xs:annotation>
     </xs:attribute>
+    <xs:attribute name="meta_file_key" type="xs:string">
```
Comment (mvdbeek) [inline lib/galaxy/tool_util/xsd/galaxy.xsd]: Can you separate out the meta_file_key changes, I assume that's handled in https://github.com/galaxyproject/galaxy/pull/11832 ?

Comment (qiagu) [reply lib/galaxy/tool_util/xsd/galaxy.xsd]: It's right. Once the `meta_file_key` implemented in #11832 gets merged into Galaxy, dynamic options can be made using the metadata FileParameter, `hyper_params` specifically, of `H5MLM`.

**Comment by qiagu** [comment]
@mvdbeek Thanks for merging!

**Pull request merged**